interface/client/lib/helpers/helperFunctions.js

```diff
@@ -50,6 +50,28 @@ Helpers.formatUrl = function(url){
 };
 
 /**
+Sanatizes URLs to prevent phishing and XSS attacks
+
+@method sanatizeUrl
+@param {String} url
+**/
+Helpers.sanatizeUrl = function(url){
+    var parser;
+
+    if(_.isString(url)) {
+        parser = document.createElement('a');
+        parser.href = url;
+    } else if(_.isObject(url) && url.protocol) {
+        parser = url;
+    }
+
+    parser.protocol = parser.protocol.replace(/^(javascript:)?(data:)?(:)?/i, 'http:');
```

Comment:
See [playground example](aW5pdEFwcCh3aW5kb3cpO2FkZENvZGVNaXJyb3IodW5lc2NhcGUoJ3NhbmF0aXplVXJsJTIwJTNEJTIwZnVuY3Rpb24lMjh1cmwlMjklN0IlMEElMjAlMjAlMjAlMjAlMjB2YXIlMjBwYXJzZXIlM0IlMEElMjAlMjAlMjAlMjAlMjBzaG93JTI4JTIyVVJMJTNBJTIyK3VybCUyOSUwQSUyMCUyMCUyMCUyMCUyMCUwQSUyMCUyMCUyMCUyMCUyMGlmJTI4dHlwZW9mJTI4dXJsJTI5JTNEJTNEJTI3c3RyaW5nJTI3JTI5JTIwJTdCJTBBJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwcGFyc2VyJTIwJTNEJTIwZG9jdW1lbnQuY3JlYXRlRWxlbWVudCUyOCUyN2ElMjclMjklM0IlMEElMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjBwYXJzZXIuaHJlZiUyMCUzRCUyMHVybCUzQiUwQSUyMCUyMCUyMCUyMCUyMCU3RCUyMGVsc2UlMjBpZiUyOF8uaXNPYmplY3QlMjh1cmwlMjklMjAlMjYlMjYlMjB1cmwucHJvdG9jb2wlMjklMjAlN0IlMEElMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjBwYXJzZXIlMjAlM0QlMjB1cmwlM0IlMEElMjAlMjAlMjAlMjAlMjAlN0QlMEElMjAlMDklMjAlMEElMjAlMjAlMDlzaG93JTI4JTIyJTVCMiU1RCUyMEJlZm9yZSUyMHJlcGxhY2UlM0ElMjAlMjIrcGFyc2VyJTI5JTBBJTIwJTIwJTIwJTIwJTIwcGFyc2VyLnByb3RvY29sJTIwJTNEJTIwcGFyc2VyLnByb3RvY29sLnJlcGxhY2UlMjgvJTVFJTI4amF2YXNjcmlwdCUzQSUyOSUzRiUyOGRhdGElM0ElMjklM0YlMjglM0ElMjklM0YvaSUyQyUyMCUyN2h0dHAlM0ElMjclMjklM0IlMEElMjAlMjAlMDlzaG93JTI4JTIyJTVCMyU1RCUyMEFmdGVyJTIwcmVwbGFjZSUzQSUyMiUyMCslMjBwYXJzZXIlMjklMEElMjAlMjAlMjAlMjAlMjAlMEElMjAlMjAlMjAlMjAlMjB1cmwlMjAlM0QlMjBwYXJzZXIudG9TdHJpbmclMjglMjklM0IlMEElMjAlMEElMjAlMjAlMjAlMjAlMjByZXR1cm4lMjB1cmwlM0IlMEElMjAlN0QlM0IlMEElMEFzaG93JTI4c2FuYXRpemVVcmwlMjglMjJodHRwcyUzQS8vd3d3LmRuLnNlJTIyJTI5JTI5JykpOw0KdXBkYXRlT3V0cHV0KCk7DQo). This function will replace "https://" urls with "http://"

This is (I think) because the matches all, due to there only being optional capture groups. See [here](aW5pdEFwcCh3aW5kb3cpO2FkZENvZGVNaXJyb3IodW5lc2NhcGUoJ3NhbmF0aXplVXJsJTIwJTNEJTIwZnVuY3Rpb24lMjh1cmwlMjklN0IlMEElMjAlMjAlMjAlMjAlMjB2YXIlMjBwYXJzZXIlM0IlMEElMjAlMjAlMjAlMjAlMjBzaG93JTI4JTIyVVJMJTNBJTIyK3VybCUyOSUwQSUyMCUyMCUyMCUyMCUyMCUwQSUyMCUyMCUyMCUyMCUyMGlmJTI4dHlwZW9mJTI4dXJsJTI5JTNEJTNEJTI3c3RyaW5nJTI3JTI5JTIwJTdCJTBBJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwcGFyc2VyJTIwJTNEJTIwZG9jdW1lbnQuY3JlYXRlRWxlbWVudCUyOCUyN2ElMjclMjklM0IlMEElMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjBwYXJzZXIuaHJlZiUyMCUzRCUyMHVybCUzQiUwQSUyMCUyMCUyMCUyMCUyMCU3RCUyMGVsc2UlMjBpZiUyOF8uaXNPYmplY3QlMjh1cmwlMjklMjAlMjYlMjYlMjB1cmwucHJvdG9jb2wlMjklMjAlN0IlMEElMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjBwYXJzZXIlMjAlM0QlMjB1cmwlM0IlMEElMjAlMjAlMjAlMjAlMjAlN0QlMEElMjAlMDklMjAlMEElMjAlMjAlMDlzaG93JTI4JTIyJTVCMiU1RCUyMEJlZm9yZSUyMHJlcGxhY2UlM0ElMjAlMjIrcGFyc2VyJTI5JTBBJTIwJTIwJTIwJTIwJTBBJTIwJTIwJTIwJTIwaWYlMjglMjBwYXJzZXIucHJvdG9jb2wlMjAlM0QlM0QlMjAlMjJqYXZhc2NyaXB0JTNBJTIyJTIwJTdDJTdDJTBBJTIwJTIwJTIwJTIwJTIwJTIwJTIwcGFyc2VyLnByb3RvY29sJTIwJTNEJTNEJTIyZGF0YSUzQSUyMiUyOSUwQSUyMCUyMCUyMCUyMCU3QiUwQSUyMCUyMCUyMCUyMCUyMCUyMHBhcnNlci5wcm90b2NvbCUyMCUzRCUyMCUyMmh0dHBzJTNBLy8lMjIlMEElMjAlMjAlMjAlMjAlN0QlMEElMEElMjAlMjBzaG93JTI4JTIyJTVCMyU1RCUyMEFmdGVyJTIwcmVwbGFjZSUzQSUyMiUyMCslMjBwYXJzZXIlMjklMEElMjAlMjAlMjAlMjAlMjAlMEElMjAlMjAlMjAlMjAlMjB1cmwlMjAlM0QlMjBwYXJzZXIudG9TdHJpbmclMjglMjklM0IlMEElMjAlMEElMjAlMjAlMjAlMjAlMjByZXR1cm4lMjB1cmwlM0IlMEElMjAlN0QlM0IlMEElMEFzaG93JTI4c2FuYXRpemVVcmwlMjglMjJqYXZhcyU1Q3RjcmlwVCUzQWFsZXJ0JTI4MSUyOSUyMiUyOSUyOScpKTsNCnVwZGF0ZU91dHB1dCgpOw0K) for a more robust example.


And I still think that any input _not_ explicitly  supported, should be coerced into a string. Otherwise it does not get filtered.